test/e2e/storage/testsuites/base.go

```diff
@@ -192,9 +192,19 @@ func (r *genericVolumeTestResource) cleanupResource(driver drivers.TestDriver, p
 	volType := pattern.VolType
 
 	if r.pvc != nil || r.pv != nil {
-		By("Deleting pv and pvc")
-		if errs := framework.PVPVCCleanup(f.ClientSet, f.Namespace.Name, r.pv, r.pvc); len(errs) != 0 {
-			framework.Failf("Failed to delete PVC or PV: %v", utilerrors.NewAggregate(errs))
+		switch volType {
+		case testpatterns.PreprovisionedPV:
+			By("Deleting pv and pvc")
+			if errs := framework.PVPVCCleanup(f.ClientSet, f.Namespace.Name, r.pv, r.pvc); len(errs) != 0 {
+				framework.Failf("Failed to delete PVC or PV: %v", utilerrors.NewAggregate(errs))
+			}
+		case testpatterns.DynamicPV:
+			By("Deleting pvc")
+			// We only delete the PVC so that PV (and disk) can be cleaned up by dynamic provisioner
+			err := framework.DeletePVCandValidatePV(f.ClientSet, f.Namespace.Name, r.pvc, r.pv, v1.VolumeReleased)
```

Comment:
using this function here is racy.  PV will only be a released state for a short time before delete operation completes and fully deletes the PV object.

We could delete the PVC and wait for PV to be gone.

Also, if reclaim policy is Retain, then we need to manually delete PVC, PV and the underlying disk.  Maybe we should error if Retain is set for now and add support in the future if we need to.

Reply:
changed to just check reclaim policy is set to delete, then delete the PVC.